### ISSUE TITLE
How to generate and document spyder (md vs rst)

### ISSUE THREAD

**Issue by rlaverde** [spyder-ide/spyder#4275]
I open this issue because I think we need to decide some things:
- Where spyder documentation should live
- In which language should be written (`rst` or `md`)
- And where will be hosted.

At the moment spyder documentation is in the spyder repo (written in `rst`) and is hosted in [pythonhosted](https://pythonhosted.org/spyder/); and development documentation is in the [github wiki](https://github.com/spyder-ide/spyder/wiki) (writen in `md`).

Options:

## Move documentation to spyder website repo and use Lektor (md)
Pros:
- Will be integrated with the site
- Markdown is somehow simple to write than rst

Cons:
- Will live in another repo (when developing, code and docs should be changed in different PRs in different repos)
- Writing some advanced stuff (dev docs, api docs) in md could be difficult: there's an standard, we will need plugins, or embed html into the docs

There is already a try to do [this](https://spyder-ide.github.io/docs/) (and docs are duplicated in spyder and website repos.)

## Keep documentation in spyder repo and use Sphinx (rst)
Pros
- Sphinx is somehow an standard for documentation.
- We could use introspection of the code. (It will be useful for api docs)

Cons:
- We will need a way to integrate it with the site (generate html with Sphinx and then copy it to the website, maybe an webhook)

## Keep documentation in spyder repo and use Sphinx (rst) and readthedocs

Pros:
- Configuring the generation will be easy
- readthedocs allow to manage versions (maybe we could use it for stable(3.1.x) and latest(master))
- Search will be better (website search is static and based in js)

Cons:
- Will be hosted in an other site, and won't be integrate it with the website(we could customize the template to look like the website but will be still a different domain)

------
Which option do you think is the best? and will encourage to improve and use more the docs?

~NOTE: This is somehow a follow up of the issue #903~

**Comment by ccordoba12** [comment]
I'd like to move our docs to our website repo, so our users can find all our documentation in the same place. The API docs could go to Read the Docs.

@goanpeca?

**Comment by goanpeca** [comment]
Yep that is my preference for main spyder docs. For the api I would prefer to use the site as well but if we think read the docs is best, then ok

**Comment by ccordoba12** [comment]
I don't know if lector can mix rst and md. If it can, then our API docs can go to our website too.
